pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -32,38 +32,58 @@ type CassandraCluster struct {
 type CassandraClusterSpec struct {
 	NavigatorClusterConfig `json:",inline"`
 
-	NodePools []CassandraClusterNodePool `json:"nodePools"`
+	// List of node pools belonging to this CassandraCluster.
+	// NodePools cannot currently be removed.
+	// A cluster with 0 node pools will not function correctly.
+	NodePools []CassandraClusterNodePool `json:"nodePools,omitempty"`
 
-	// Image describes the database image to use
-	Image *ImageSpec `json:"image"`
+	// Image describes the Cassandra database image to use.
+	// This should only be set if version auto-detection is not desired.
+	// If set, the image tag used must correspond to the version specified
+	// in 'spec.version'.
+	Image *ImageSpec `json:"image,omitempty"`
 
 	// The version of the database to be used for nodes in the cluster.
+	// This field must be a valid Cassandra version, e.g. '3.11.2'.
 	Version version.Version `json:"version"`
 }
 
 // CassandraClusterNodePool describes a node pool within a CassandraCluster.
 type CassandraClusterNodePool struct {
-	Name     string `json:"name"`
-	Replicas int32  `json:"replicas"`
+	// Name of the node pool specified as a DNS_LABEL.
+	// Each node pool must specify a name.
+	Name string `json:"name"`
+
+	// The number of desired replicas in this node pool.
+	// This value will correspond to the number of replicas in the created
+	// StatefulSet.
+	// If not set, a default of 1 will be used.
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Also not sure about this being `omitempty`.
It seems to me that it'd be better to require this to be specified.
Can / should these be uint32 to ensure that they're not negative?

Reply:
We have omitempty here again because otherwise this will be serialised to null. Defaults will handle setting this to an appropriate value if not specified.

Having this as a pointer allows us to recognise `nil` vs `0` (so we can support 0 sized node pools). I chose to do this because every other Kubernetes workloads controller treats 'nil' to mean 1 (i.e., not specified and thus defaulting to 1).

This also helps us work with autoscalers in future (as the best practice when using an autoscaler is to not specify a replicas field in VCS, and instead allow the autoscaler to set it). If we made this a non-ptr/required, whenever a `kubectl apply` is performed, the `replicas` field would be reset to 0.